exercises/03-exercise.py

```diff
@@ -0,0 +1,121 @@
+''' 
+Author: Jorge Diaz Jr
+Exercise 3
+'''
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+
+file_path = "/home/jdiaz/projects/data-monitoring/data/xas_data/xas_test_328.txt"
+
+def get_header(file_path):
+	'''
+	Open file , reads file in, splits line apart if it starts with #, 
+	one piece is the key, the other piece is the value
+	e.g. # Year: 2018
+	gets stored in a dictionary such that hdr_dict['Year'] = '2018'
+	returns the dictionary hdr_data, and returns a list of all the lines
+	such that the lines do NOT start with # symbol
+	'''
+	hdr_data = {}
+	key_value = []
+	info_data = []
+
+	#opens file and reads each line, extracting key and value pair if line starts with '#'
+	with open(file_path) as file:
+		for line in file:
+			if line.startswith("#"):
+				if len(line) > 1:
+					key_value = line.split(":", 1)
+					if len(key_value) == 2:
+						key = key_value[0].replace("#","").strip()
+						value = key_value[1].replace("\n","").strip()
+						hdr_data[key] = value
+					elif len(key_value) == 1:
+						continue	
+			if not line.startswith("#"):
+				info_data.append(line.strip().split())
+
+	return hdr_data, info_data
+
+def get_column_names(file_path):
+	'''
+	argument in function is file path
+	this function reads the lines to extract
+	column names and returns them as a list
+	'''
+	column_names = []
+	with open(file_path) as file:
+		for line in file:
+			if line.startswith("#"):
+				if len(line) > 1:
+					key_value = line.split(":", 1)
+					if len(key_value) == 1:
+						column_names = line.replace("#","").strip().split()
+						continue
+	return column_names
+
+def create_df(info_data, column_names):
+	'''
+	the list made up of lines is passed in as well as column names
+	creates a dataframe object and returns it
+	'''
+	df = pd.DataFrame(info_data)
+	df.columns = column_names
+	print("\nData Frame")
+	print(df)
+	return df 
+
+def plot_log_i0_it(i0_data, it_data):
```

Comment:
looks good. something useful could be passing figure number as a keyword argument.

A common pattern is:
```py3
def plot_log_i0_it(i0_data, it_data, fignum=None):
    if fignum is None:
        fignum = 0
plt.figure(fignum)
```

Here, we use `None` as an identifier that we want to let the function specify it's own default. We could have just set the default to `fignum=0`. Why didn't I do that?
One reason is that as code gets complex and functions call other functions etc, it is easier to remember to type `None` when a default is needed than have to remember the defaults. Another reason, is it allows the user of the function to change the default if necessary.